app/javascript/components/admin/EmailModal.js

```diff
@@ -16,9 +16,15 @@ class EmailModal extends React.Component {
 
   render() {
     return (
-      <Modal show={this.props.show} onHide={this.props.onClose} backdrop="static" aria-labelledby="contained-modal-title-vcenter" centered>
+      <Modal
+        data-testid="email_modal"
+        show={this.props.show}
+        onHide={this.props.onClose}
+        backdrop="static"
+        aria-labelledby="contained-modal-title-vcenter"
+        centered>
         <Modal.Header closeButton>
-          <Modal.Title>{this.props.title}</Modal.Title>
+          <Modal.Title data-testid="email_title">{this.props.title}</Modal.Title>
```

Comment:
This doesn't need a `data-testid` attribute, nor do any of the other places it's included really because the content should be known when rendering in the test (since it's just based on props). I would avoid using this unless really necessary, like they've outlined [here](https://kentcdodds.com/blog/making-your-ui-tests-resilient-to-change).

Reply:
I removed all the ones testing on props but left the rest in.  there isn't a good other way to do it (i.e. test by class or something). if you think more need to be removed let me know which ones

Reply:
So I actually don't think those are necessary either - with the use of getByText(), getByDisplayValue(), and getByTitle(). 
https://testing-library.com/docs/dom-testing-library/api-queries#bytext